File: app/src/main/java/me/ykrank/s1next/widget/track/TrackAgent.java
package me.ykrank.s1next.widget.track;

import android.app.Activity;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.Map;

import me.ykrank.s1next.data.User;

/**
 * Created by ykrank on 2016/12/29.
 */

public interface TrackAgent {
    void init();

    void setUser(@Nullable User user);

    void onResume(@NonNull Activity activity);

    void onPause(@NonNull Activity activity);

    void onPageStart(@NonNull Context context, String string);

    void onPageEnd(@NonNull Context context, String string);

    void onEvent(@NonNull Context context, String name, String label, Map<String, String> data);
}

File: library/src/main/java/com/github/ykrank/androidtools/widget/track/TrackAgent.kt
package com.github.ykrank.androidtools.widget.track

import android.app.Activity
import android.content.Context

import com.github.ykrank.androidtools.data.TrackUser

/**
 * Created by ykrank on 2016/12/29.
 */

interface TrackAgent {
    fun init(context: Context)

    fun setUser(user: TrackUser)

    fun onResume(activity: Activity)

    fun onPause(activity: Activity)

    fun onPageStart(context: Context, string: String)

    fun onPageEnd(context: Context, string: String)

    fun onEvent(context: Context, name: String, label: String, data: Map<String, String?>)
}
